scripts/skills/perks/perk_rf_bloodlust.nut

```diff
@@ -22,82 +22,90 @@ this.perk_rf_bloodlust <- ::inherit("scripts/skills/skill", {
+			type = "text",
+			icon = "ui/icons/initiative.png",
+			text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorizeMult(mult) + " increased [Initiative|Concept.Initiative]")
+		});
+		ret.push({
+			id = 12,
+			type = "text",
 			icon = "ui/icons/fatigue.png",
-			text = "[color=" + ::Const.UI.Color.PositiveValue + "]+" + this.m.FatigueRecoveryStacks + "[/color] Fatigue Recovery on the next turn"
+			text = ::Reforged.Mod.Tooltips.parseString("Skills build up " + ::MSU.Text.colorizeMult(mult) + " less [Fatigue Cost|Concept.Fatigue]")
+		});
+		ret.push({
+			id = 13,
+			type = "text",
+			icon = "ui/tooltips/warning.png",
+			text = ::Reforged.Mod.Tooltips.parseString("These effects are reduced by -" + (this.m.MultPerStack * 100) "% every [turn|Concept.Turn] until another fatality")
 		});
-		return tooltip;
+		return ret;
 	}
 
-	function onBeforeAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	function onOtherActorDeath( _killer, _victim, _skill, _deathTile, _corpseTile, _fatalityType )
 	{
-		this.m.ActorFatigue = null;
-		this.m.BleedStacksBeforeAttack = 0;
-		this.m.DidHit = false;
-
-		if (_skill.isAttack() && !_skill.isRanged() && _targetEntity != null)
+		if (_fatalityType != ::Const.FatalityType.None && _killer != null && _killer.getID() == this.getContainer().getActor().getID() && _skill != null && this.isSkillValid(_skill))
 		{
-			this.m.BleedStacksBeforeAttack = _targetEntity.getSkills().getAllSkillsByID("effects.bleeding").len();
+			this.m.Stacks = ::Math.min(this.m.MaxStacks, this.m.Stacks + this.m.StacksPerTrigger);
 		}
 	}
 
-	function onTargetHit( _skill, _targetEntity, _bodyPart, _damageInflictedHitpoints, _damageInflictedArmor )
-	{
-		this.m.DidHit = true;
-	}
-
-	// We need to do it like this in two split functions i.e. onAnySkillExecuted and onTargetKilled:
-	// onAnySkillExecuted is used for counting the bleed stacks applied during the attack e.g. cleave skill's onUse function
-	// onTargetKilled is used separately to handle situations where the attack is delayed e.g. Lunge, and onAnySkillExecuted runs before the target actually gets killed
-	function onAnySkillExecuted( _skill, _targetTile, _targetEntity, _forFree )
+	function onUpdate( _properties )
 	{
-		local actor = this.getContainer().getActor();
-		if (!this.m.DidHit || !_skill.isAttack() || _skill.isRanged() || _targetEntity == null || _targetEntity.isAlliedWith(actor) || !::Tactical.TurnSequenceBar.isActiveEntity(actor))
+		if (this.m.Stacks == 0)
 			return;
 
-		local bleedCount = _targetEntity.isAlive() ? _targetEntity.getSkills().getAllSkillsByID("effects.bleeding").len() : this.m.BleedStacksBeforeAttack;
-
-		this.m.FatigueRecoveryStacks += bleedCount;
-
-		if (this.m.ActorFatigue == null) this.m.ActorFatigue = actor.getFatigue();
-
-		actor.setFatigue(::Math.max(0, this.m.ActorFatigue - this.m.ActorFatigue * (bleedCount * this.m.FatigueReductionPercentage * 0.01)));
+		local mult = this.m.MultPerStack * this.m.Stacks;
```

Comment:
I would move the +1 from the lines below up in this line. A `mult` imo is something that is already ready to be multiplied with another value and need no further adjustment.

So like this
`local mult = 1.0 + (this.m.MultPerStack * this.m.Stacks);`